RoaringBitmap/src/test/java/org/roaringbitmap/longlong/TestRoaring64Bitmap.java

```diff
@@ -1739,6 +1739,201 @@ public void testEmptySkipsReverse() {
+    Roaring64Bitmap bitset = new Roaring64Bitmap();
+    long b1 = 2000000000L;
+    long b1s = 18500L;
+    long b1e = b1 + b1s;
+    long b2 = 4000000000L;
+    long b2s = 100L;
+    long b2e = b2 + b2s;
+    long p2 = b2 + (b2s / 2);
+    long pgap = p2 - b1s;
+
+    bitset.add(b1, b1e);
+    bitset.add(b2, b2e);
+
+    PeekableLongIterator bitIt = bitset.getReverseLongIterator();
+
+    assertEquals(b2e - 1L, bitIt.peekNext());
+    assertEquals(b2e - 1L, bitIt.next());
+
+    assertTrue(bitset.contains(p2));
+    bitIt.advanceIfNeeded(p2);
+    assertEquals(p2, bitIt.peekNext());
+    assertEquals(p2, bitIt.next());
+
+    // advancing to a value not in either range should go to the first value of second range
+    assertFalse(bitset.contains(pgap));
+    bitIt.advanceIfNeeded(pgap);
+
+    assertTrue(bitset.contains(b1));
+    assertTrue(bitset.contains(b1e - 1L));
+    assertEquals(b1e - 1L, bitIt.peekNext());
+
+    assertTrue(bitset.contains(b2));
+    bitIt.advanceIfNeeded(b1e);
+    assertEquals(b1e - 1L, bitIt.peekNext());
+    assertEquals(b1e - 1L, bitIt.next());
+  }
+
+  @Test
+  public void testSkipIntoFarAwayGapsReverse() {
+    Roaring64Bitmap bitset = new Roaring64Bitmap();
+    long runLength = 18500L;
+    long b1 = 2000000000L;
+    long b1e = b1 + runLength;
+    long b2 = 4000000000L;
+    long b2e = b2 + runLength;
+    long p3 = b2 + (runLength / 2);
+    long pgapSameContainer = p3 - runLength;
+    long pgapNextContainer = p3 - 5 * runLength;
+    long b3 = 6000000000L;
+    long b3e = b3 + runLength;
+
+    bitset.add(b1, b1e);
+    bitset.add(b2, b2e);
+    bitset.add(b3, b3e);
+
+    PeekableLongIterator bitIt = bitset.getReverseLongIterator();
+
+    assertEquals(b3e - 1L, bitIt.peekNext());
+    assertEquals(b3e - 1L, bitIt.next());
+
+    assertTrue(bitset.contains(p3));
+    bitIt.advanceIfNeeded(p3);
+    assertEquals(p3, bitIt.peekNext());
+    assertEquals(p3, bitIt.next());
+
+    // advancing to a value not in any range but beyond second range
+    // should go to the first value of third range
+    assertFalse(bitset.contains(pgapSameContainer));
+    bitIt.advanceIfNeeded(pgapSameContainer);
+
+    assertTrue(bitset.contains(b1));
+    assertTrue(bitset.contains(b1e - 1L));
+    assertEquals(b1e - 1L, bitIt.peekNext());
+
+    assertTrue(bitset.contains(b1));
+    bitIt.advanceIfNeeded(b1e);
+    assertEquals(b1e - 1L, bitIt.peekNext());
+    assertEquals(b1e - 1L, bitIt.next());
+
+    // reset
+    bitIt = bitset.getReverseLongIterator();
+    bitIt.advanceIfNeeded(p3);
+
+    // advancing to a value not in any range but beyond second range
+    // should go to the first value of third range
+    assertFalse(bitset.contains(pgapNextContainer));
+    bitIt.advanceIfNeeded(pgapNextContainer);
+
+    assertEquals(b1e - 1L, bitIt.peekNext());
+
+    bitIt.advanceIfNeeded(b1e);
+    assertEquals(b1e - 1L, bitIt.peekNext());
+    assertEquals(b1e - 1L, bitIt.next());
+  }
```

Comment:
at this point in the test I wondered "what happens if I advance past the last value" so insert some more code
```
    System.out.println(bitIt.peekNext());
    System.out.println(b3e);
    bitIt.advanceIfNeeded(b3e);
    assertFalse(bitset.contains(b3e));
    System.out.println(bitIt.peekNext());
    assertFalse(bitIt.hasNext());
```

surprising to me the assert fails. with the sout's I show these values:
```
2000018498
6000018500
2000018498
```

which makes sense from the perspective of there is "no such value, thus no where to advance to" but the example in the abstract class is a merge join, and that would break if the bitmap iterator fails to advance to the end, and thus would spin forever. I didn't test this with the 32bit code, nor on the original code, to see if it is a side effect of the change.

Reply:
Is there some action item here, or I can I mark this as resolved?

Reply:
resolved.